pkg/agent/route/route_linux.go

```diff
@@ -520,10 +567,11 @@ func (c *Client) restoreIptablesData(podCIDR *net.IPNet,
 	writeLine(iptablesData, iptables.MakeChainLine(antreaMangleChain))
 	writeLine(iptablesData, iptables.MakeChainLine(antreaOutputChain))
 
-	// When Antrea is used to enforce NetworkPolicies in EKS, an additional iptables
-	// mangle rule is required. See https://github.com/antrea-io/antrea/issues/678.
-	if env.IsCloudEKS() {
-		c.writeEKSMangleRule(iptablesData)
+	// When Antrea is used to enforce NetworkPolicies in EKS, additional iptables
+	// mangle rules are required. See https://github.com/antrea-io/antrea/issues/678.
+	// These rules are only needed for IPv4.
```

Comment:
Does it have NodePort IPv6 case? If yes, is the rule still needed for IPv6, otherwise reverse path may be different?

Reply:
yes, I think you're right. This one should always be installed.
Note that we don't test Antrea on EKS with IPv6 enabled, so there may be other issues, but I'll fix that one.

Reply:
Actually scratch that. The AWS rules are only installed for IPv4: https://github.com/aws/amazon-vpc-cni-k8s/blob/d43309bdfdb5034df86907944e682d78608ba165/pkg/networkutils/network.go#L402-L418. Not entirely sure why, but I would think that routing is configured differently for IPv6, and these rules are not needed.

Reply:
There is this comment:

```
		//Essentially a stub function for now in V6 mode. We will need it when we support v6 in secondary IP and
		//custom networking modes. We don't need to install any SNAT rules in v6 mode and currently there is no need
		//to mark packets entering via Primary ENI as all the pods in v6 mode will be behind primary ENI. Will have to
		//start doing that once we start supporting custom networking mode in v6.
```

So that explains why there is no need for the rule: only the primary ENI is used, with no secondary route table.

Reply:
got it, thanks